extensions/interactions/NumericInput/directives/numeric-input-validation.service.ts

```diff
@@ -207,9 +207,37 @@ export class NumericInputValidationService {
   }
 
   // Returns 'undefined' when no error occurs.
-  getErrorStringI18nKey(
-      value: number, requireNonnegativeInput: boolean
+  validateNumericString(
+      value: string, decimalSeparator: string): string | undefined {
+    value = value.toString().trim();
+    const trailingDot = /[\.|\,|\u066B]\d/g;
+    const twoDecimals = /.*[\.|\,|\u066B].*[\.|\,|\u066B]/g;
+    const trailingMinus = /(^-)|(e-)/g;
+    const extraMinus = /-.*-/g;
+    const extraExponent = /e.*e/g;
+
+    if (value.includes(decimalSeparator) && !value.match(trailingDot)) {
+      return 'I18N_INTERACTIONS_NUMERIC_INPUT_NO_TRAILING_DECIMAL';
+    } else if (value.match(twoDecimals)) {
+      return 'I18N_INTERACTIONS_NUMERIC_INPUT_ATMOST_1_DECIMAL';
+    } else if (value.includes('-') && !value.match(trailingMinus)) {
+      return 'I18N_INTERACTIONS_NUMERIC_INPUT_MINUS_AT_BEGINNING';
+    } else if (value.includes('-') && value.match(extraMinus)) {
+      return 'I18N_INTERACTIONS_NUMERIC_INPUT_ATMOST_1_MINUS';
+    } else if (value.includes('e') && value.match(extraExponent)) {
+      return 'I18N_INTERACTIONS_NUMERIC_INPUT_ATMOST_1_EXPONENT';
+    }
+  }
+
+  // Returns 'undefined' when no error occurs.
+  validateNumber(
+      value: number, requireNonnegativeInput: boolean,
+      decimalSeparator: string = '.'
```

Comment:
So we want dot as a default value here?

Reply:
Yup! The default language should be English, therefore we are using '.' as the default decimal separator.